install_ffmpeg.sh

```diff
@@ -41,6 +56,7 @@ if [ ! -e "$HOME/ffmpeg/libavcodec/libavcodec.a" ]; then
     --enable-filter=aresample,asetnsamples,fps,scale \
     --enable-encoder=aac,libx264 \
```

Comment:
That reminds me, I wonder if there are additional flags we'll need to add for a minimal build:

* h264_nvenc encoder
* scale_npp, hwupload_cuda and hwdownload filters
* Anything for the decoder hwaccel?

Rather than just add these in blindly, let's test this build in practice and see what the results look like, so we aren't cargo-culting unnecessary flags. I've been testing with a fully loaded ffmpeg and forgot that we might need to make updates to the minimal build.

Reply:
Looked at this a bit more and this is what I came up with.

The `--extra-cflags` and `--extra-ldflags` are needed here because CUDA installs itself in a nonstandard location but may not be required for Appveyor builds.

I also have to build the goclient with the `LIBRARY_PATH=/usr/local/cuda/lib64` environment variable when ffmpeg is linked statically (as is here).

I'm also unsure sure if we should actually add these options by default to the base build, since a lot of users won't actually be GPU enabled and we don't want to foist unnecessary dependencies on them. But we should definitely enable it for CI. Maybe toggle via environment variable or checking the presence of `/usr/local/cuda` ?

```diff
diff --git a/install_ffmpeg.sh b/install_ffmpeg.sh
index f47b5508..59ec8c3e 100755
--- a/install_ffmpeg.sh
+++ b/install_ffmpeg.sh
@@ -32,15 +32,18 @@ if [ ! -e "$HOME/ffmpeg/libavcodec/libavcodec.a" ]; then
     --disable-muxers --disable-demuxers --disable-parsers --disable-protocols \
     --disable-encoders --disable-decoders --disable-filters --disable-bsfs \
     --disable-postproc --disable-lzma \
-    --enable-gnutls --enable-libx264 --enable-gpl \
+    --enable-gnutls --enable-libx264 --enable-gpl --enable-nonfree \
     --enable-protocol=https,rtmp,file \
     --enable-muxer=mpegts,hls,segment --enable-demuxer=flv,mpegts \
     --enable-bsf=h264_mp4toannexb,aac_adtstoasc,h264_metadata,h264_redundant_pps \
     --enable-parser=aac,aac_latm,h264 \
     --enable-filter=abuffer,buffer,abuffersink,buffersink,afifo,fifo,aformat \
-    --enable-filter=aresample,asetnsamples,fps,scale \
-    --enable-encoder=aac,libx264 \
+    --enable-filter=aresample,asetnsamples,fps,scale,scale_npp \
+    --enable-encoder=aac,libx264,h264_nvenc \
     --enable-decoder=aac,h264 \
+    --enable-cuda --enable-cuvid --enable-nvenc --enable-libnpp \
+    --extra-cflags=-I/usr/local/cuda/include \
+    --extra-ldflags=-L/usr/local/cuda/lib64 \
     --prefix="$HOME/compiled"
   make
   make install
```